crates/sage/src/database.rs

```diff
@@ -84,6 +84,10 @@ pub struct Builder {
     pub generate_decoys: Option<bool>,
     /// Path to fasta database
     pub fasta: Option<String>,
+    /// Number of sequences to handle simultaneously when pre-filtering the db
```

Comment:
A super cool function would be if `prefilter=true` and `prefilter_chunk_size=0` (or -1, etc.). to automatically guesstimate a somewhat reasonable chunk size based on a) 80-90% of system memory, b) the number of peptides, which digestion of proteins 1-10 generate.

This is not a requirement for initial release but more a task for later or if someone is interested in briefly looking into this.

Reply:
Why this would be useful: Memory usage is quite dependent on number of peptides that need to be searched per protein. Having such a dynamic function in dependency of available memory would engineer away potential issues also for PTM search, very large DBs, etc. and ensure that runtime is always ideal.

Reply:
Agreed, I thought I actually left a comment in the fasta.iter_chunks as that might be the sensible point to put it in, but sems like I deleted that:P